databricks_cli/configure/provider.py

```diff
@@ -185,20 +185,41 @@ def get_config(self):
 
 
 class DefaultConfigProvider(DatabricksConfigProvider):
-    """Prefers environment variables, and then the default profile."""
+    """Look for credentials in a chain of default locations."""
     def __init__(self):
-        self.env_provider = EnvironmentVariableConfigProvider()
-        self.default_profile_provider = ProfileConfigProvider()
+        self._providers = (
+            SparkTaskContextConfigProvider(),
+            EnvironmentVariableConfigProvider(),
+            ProfileConfigProvider()
+        )
 
     def get_config(self):
-        env_config = self.env_provider.get_config()
-        if env_config:
-            return env_config
+        for provider in self._providers:
```

Comment:
Do we have a test for this logic?

Reply:
The logic is the same as before except there is an extra provider at the start of the chain . 

I think the existing tests + the the new test for the new provider added in this PR should suffice. 
We do not test the order of the providers. I don't think that requires a dedicated test, I can add one though if you think we should have it.